### PR TITLE
Module api

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?

Add module API  MD doc

## How was this patch tested?

Tested locally for all the example code

## Related links or issues (optional)
https://github.com/intel-analytics/BigDL/issues/1203

### CODE REVIEW

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+```
+###### Save model to specific path
+
+```python
+model.save("/tmp/model.bigdl", True)
+```
+###### Load model from given path
+
+```python
+model = Model.load("/tmp/model.bigdl")
```
Comment (shane-huang) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: can we add the import libraries here so that user will know which lib `Model` belongs to?

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+Create dummy test data set 
+
+```python
+samples=[Sample.from_ndarray(np.array([1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0]), np.array([2.0]))]
+
+testSet = sc.parallelize(samples)
+
+```
+`np` above is numpy package
```
Comment (shane-huang) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: you can just use `import numpy as np` instead of describe what `np` is.

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+Predict with the data set
+```scala
+val preductResult = model.predict(predictSet)
+```
+
+**Python:**
+
+###### Define the model
+```python
+model = Sequential().add(Linear(10, 5)).add(Sigmoid()).add(SoftMax())
```
Comment (shane-huang) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: better add some `import...`

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+Use the test dataset to evaluate the model
+
+```scala
+val evaluateResult = model.evaluate(testSet, Array(new Top1Accuracy))
+```
+
+###### Predict with model
+Similar to above, we create input data first
+```scala
```
Comment (shane-huang) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: can we add some `import` sentences so that user will know which libs to import to use these API's?

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+Create dummy test data set
+
+```scala
+val feature = Tensor(10).apply1(e => Random.nextFloat())
+
+val label = Tensor(1).apply1(e => Random.nextFloat())
+
+val testSample = Sample(feature, label)
+    
+val testSet : RDD[Sample[Float]] = sc.parallelize(Seq(testSample))
```
Comment (shane-huang) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: remove [Float]?

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+```
+
+In the above code , the first parameter is the path where we want to save our model, the second paramter is to specify if we need to overwrite the file if it already exists, it's set to false by default
+
+###### Load model from given path
+
+We just need to specify the model path where we previously saved the model to load it to memory for resume training or prediction purpose
+
+```scala
+val loadedModel = Module.load("/tmp/model.bigdl")
```
Comment (shane-huang) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: Maybe some `import ...`?

**Comment by wzhongyuan** [comment]
@shane-huang  I refined the file per your comments, please help check. Thanks

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+`Module` above is a utilily  to manipulate module APIs
+
+###### Evaluate model with test set
+
+Evaluation needs real test data set, here we just simulate some dummy data for function introduction purpose
+
+Create dummy test data set
+
+```scala
+val feature = Tensor(10).apply1(e => Random.nextFloat())
```
Comment (yiheng) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: Tensor(10).rand ?

File: docs/docs/APIdocs/Model/ModuleAPI.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+val label = Tensor(1).apply1(e => Random.nextFloat())
```
Comment (yiheng) [inline docs/docs/APIdocs/Model/ModuleAPI.md]: Tensor(1).random()?

**Comment by wzhongyuan** [comment]
@yiheng  Modified.

**Comment by yiheng** [comment]
Looks like the PR cannot be rebased...

**Comment by wzhongyuan** [comment]
@yiheng  there should be no problem as it's a newly added file.

**Comment by wzhongyuan** [comment]
there is some issues with my local files, I will create another one and merge directly after it's resolved @yiheng
